app/controllers/idv/otp_delivery_method_controller.rb

```diff
@@ -50,8 +42,43 @@ def otp_delivery_selection_params
       )
     end
 
+    def send_phone_confirmation_otp
+      save_delivery_preference_in_session
+      result = send_phone_confirmation_otp_service.call
+      analytics.track_event(Analytics::IDV_PHONE_CONFIRMATION_OTP_SENT, result.to_h)
+      if send_phone_confirmation_otp_service.user_locked_out?
+        handle_too_many_otp_sends
+      else
+        redirect_to idv_otp_verification_url
+      end
+    end
+
+    def save_delivery_preference_in_session
+      idv_session.phone_confirmation_otp_delivery_method =
+        @otp_delivery_selection_form.otp_delivery_preference
+    end
+
+    def send_phone_confirmation_otp_service
+      @send_phone_confirmation_otp_service ||= Idv::SendPhoneConfirmationOtp.new(
+        user: current_user,
+        idv_session: idv_session,
+        locale: user_locale
+      )
+    end
+
+    def user_locale
+      available_locales = PhoneVerification::AVAILABLE_LOCALES
+      http_accept_language.language_region_compatible_from(available_locales)
+    end
+
     def otp_delivery_selection_form
       @otp_delivery_selection_form ||= Idv::OtpDeliveryMethodForm.new
     end
+
+    def invalid_phone_number(exception)
```

Comment:
Thanks! I think we also want to copy over the `capture_analytics_for_exception(exception)` method to track these errors.